pyvista/core/composite.py

```diff
@@ -1110,6 +1110,8 @@ def as_polydata_blocks(self, copy=False):
             if block is not None:
                 if isinstance(block, MultiBlock):
                     dataset.replace(i, block.as_polydata_blocks(copy=copy))
+                elif isinstance(block, pyvista.PointSet):
+                    dataset.replace(i, block.cast_to_polydata(deep=copy))
```

Comment:
My only question is about the subtlety of the `copy`/`deep` parameter. The docstring says `copy` specifies whether `PolyData` inputs get shallow copied or not. Whereas `cast_to_polydata()`'s `deep` kwarg presumably always copies, the question is whether the copy is deep or shallow.

Since for non-PolyData meshes we use `block.extract_surface()` which presumably gives a new mesh (does it?), shouldn't we use `deep=True` for the `PointSet` case?

Reply:
Extract surface indeed returns a new mesh.

I would prefer respecting the `copy` parameter as much as possible since we already do this for for existing datasets, but since non-polydata datasets are already copied, I agree that it's better to always copy.

Reply:
Hm. Should we raise an error in that filter for PointSets like other geometry/cell-wise filters? This filter seems like a no-op